### PR TITLE
[WP #54017] Update php version for api and unit test

### PR DESCRIPTION
**Pull request by nabim777**
## Description

This PR add the support for our application for php version `8.2` and `8.3`. Previously it was supporting only `8.0` and `8.1`
## Related Issue or Workpackage

- https://community.openproject.org/work_packages/54017/activity

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Updated `CHANGELOG.md` file

### CODE REVIEW

**Review by SagarGi — COMMENTED** [review]
Can you also write something in `CHANGELOG.md` regarding this PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 - Drop application's support for Nextcloud 25
+- add the php version 8.3, 8.4 in unit test and update php version from 8.1 to 8.32 in API test
```
Comment (SagarGi) [inline CHANGELOG.md:9]:
```suggestion
- Add support for PHP version 8.2 and 8.3
```

**Review by SagarGi — COMMENTED** [review]
Also please create a WP to run API tests for php 8.1 and 8.3 on nightly build only.

**Review by individual-it — CHANGES_REQUESTED** [review]
the API tests run only with PHP 8.2 currently, so ether the description of the PR is wrong or the implementation

File: tests/lib/Controller/OpenProjectAPIControllerTest.php [hunk 2]
```diff
@@ -9,25 +9,25 @@
 
 namespace OCA\OpenProject\Controller;
 
-use _PHPStan_3e014c27f\Nette\InvalidArgumentException;
```
Comment (individual-it) [inline tests/lib/Controller/OpenProjectAPIControllerTest.php:12]: good catch, how did that even get into the code :see_no_evil:

**Review by individual-it — APPROVED** [review]

**Review by SagarGi — APPROVED** [review]
LGTM if CI is green. :+1:

**Pull request merged**